### PR TITLE
Add Eslint & Prettier 

### PR DESCRIPTION
**Pull request by flashdesignory**
# Overview
Eslint rules are based on [WebInspectorUI's Eslint implementation](https://github.com/WebKit/WebKit/blob/main/Source/WebInspectorUI/.eslintrc), with a few adjustments (Envs, Globals) to ensure we adapt to Speedometer's needs. 

Prettier has been added to establish a code formatting pattern. Rules have been adjusted to avoid conflicts with the already established rules of the eslintrc file. 

This is the base configuration for both tools, with the goal to avoid additional packages to "glue the two tools together". 
Both can be used separately, with Eslint focusing on code style of JavaScript and Prettier focusing on code formatting for all supported file formats. 

# File changes
The format script (from below) has been run on the codebase.
There are two instances, where I wasn't sure if the changes might have negative impact and therefore I ignored these.
Screen Shot 2023-02-02 at 4 40 30 PM

# Scripts
The following scripts have been added to the package.json file:

```
"lint:check": "eslint **/*.mjs",
"lint:fix": "eslint **/*.mjs --fix",
"pretty:check": "prettier --check ./",
"pretty:fix": "prettier --write ./",
"format": "npm run pretty:fix ; npm run lint:fix",
```

Note: the format script runs both sequentially, even if Prettier's run results in errors. 

# CI
The format script has also been added to the Github workflow and currently runs prior to running our tests.
It is an additional job, mainly since it only needs to run once, and not in all browser we run the tests with.

# Eslint Rules
Please let me know if we want to adjust any rules, some candidates might be:
- object-curly-spacing (do we want a space between?)
- prefer-template (it's a newer convention)

### CODE REVIEW

File: resources/benchmark-report.mjs [hunk 1]
```diff
@@ -1,83 +1,107 @@
...
+    if (
+        !window.testRunner &&
+        location.search !== "?webkit" &&
+        location.hash !== "#webkit"
+    )
         return;
 
-    if (window.testRunner)
-        testRunner.waitUntilDone();
+    if (window.testRunner) window.testRunner.waitUntilDone();
```
Comment (rniwa) [inline resources/benchmark-report.mjs:11]: This is not right either. Each statement should be on its own line:
https://webkit.org/code-style-guidelines/#linebreaking-multiple-statements

File: resources/benchmark-report.mjs [hunk 2]
```diff
@@ -1,83 +1,107 @@
...
         },
         didFinishLastIteration() {
-            document.head.removeChild(document.querySelector('style'));
+            document.head.removeChild(document.querySelector("style"));
 
             const measuredValuesByFullName = {};
             function addToMeasuredValue(value, fullName, aggregator) {
-                const values = measuredValuesByFullName[fullName] || new Array;
+                const values =
+                    measuredValuesByFullName[fullName] || new Array();
```
Comment (rniwa) [inline resources/benchmark-report.mjs:57]: What's up with this?
Are we trying to set 80 character limit or something.
We certainly shouldn't.

File: resources/benchmark-runner.mjs [hunk 3]
```diff
@@ -5,15 +5,12 @@ export class BenchmarkTestStep {
     }
 }
 
-
 class Page {
-    constructor(frame)
-    {
+    constructor(frame) {
```
Comment (rniwa) [inline resources/benchmark-runner.mjs:9]: This is not right either. Curly braces should be on their own line:
https://webkit.org/code-style-guidelines/#braces-function

File: resources/benchmark-runner.mjs [hunk 4]
```diff
@@ -29,38 +26,33 @@ class Page {
...
-        if (element === null)
-            return null;
+        if (element === null) return null;
         return this._wrapElement(element);
     }
 
-    querySelectorAll(selector)
-    {
-        const elements = Array.from(this._frame.contentDocument.querySelectorAll(selector));
+    querySelectorAll(selector) {
```
Comment (housseindjirdeh) [inline resources/benchmark-runner.mjs:35]: +1 to @rniwa's comment earlier. We'll want to make sure curly braces are in a new line to adhere to https://webkit.org/code-style-guidelines/

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -1,33 +1,51 @@
 name: CI
 
 on:
-  push:
-    branches: [ "main" ]
-  pull_request:
-    branches: [ "main" ]
+    push:
```
Comment (bgrins) [inline .github/workflows/test.yml:4]: Is this file formatted by prettier / eslint? I would have expected no changes here

Comment (flashdesignory) [reply .github/workflows/test.yml:4]: we can control what files get changed. 
I believe it was prettier that changed this file

Comment (bgrins) [reply .github/workflows/test.yml:4]: Alright. I'm fine with just letting it do its thing

File: InteractiveRunner.html [hunk 6]
```diff
@@ -1,22 +1,39 @@
...
+                height: 600px;
+            }
+            nav > ol {
+                height: 100%;
+                overflow-y: scroll;
+            }
+        </style>
+    </head>
+    <body>
+        <script></script>
```
Comment (rniwa) [inline InteractiveRunner.html:37]: I guess we can get rid of this empty script element? Can be done in a separate PR though.

Comment (flashdesignory) [reply InteractiveRunner.html:37]: might as well do it now 😄

**Review by bgrins — APPROVED** [review]
Thanks Thorsten, this looks good to me

File: resources/benchmark-runner.mjs (hunk 3 above)
Comment (rniwa) [reply resources/benchmark-runner.mjs:9]: So are we reverting these curly brace changes??

**Review by housseindjirdeh — APPROVED** [review]

File: resources/benchmark-runner.mjs (hunk 3 above)
Comment (rniwa) [reply resources/benchmark-runner.mjs:9]: As I mentioned during today's sync, we're okay with adopting K&R style curly braces everywhere.

File: tests/benchmark-runner-tests.mjs [hunk 7]
```diff
@@ -1,7 +1,6 @@
-import { BenchmarkRunner } from "../resources/benchmark-runner.mjs";
+import {BenchmarkRunner} from "../resources/benchmark-runner.mjs";
```
Comment (rniwa) [inline tests/benchmark-runner-tests.mjs:1]: I think we want to keep a space after { & before }.

Comment (flashdesignory) [reply tests/benchmark-runner-tests.mjs:1]: awesome - agreed - it's more readable. 
I'll adjust the rule!

File: resources/interactive.mjs [hunk 8]
```diff
@@ -211,41 +199,34 @@ function createUIForSuites(suites, onStep, onRunSuites)
...
+        disableAllSuitesExcept(searchParams.get("suite"));
 
-    const interactiveRunner =  new window.BenchmarkRunner(Suites, params.iterationCount);
+    const interactiveRunner = new window.BenchmarkRunner(Suites, params.iterationCount);
 
     if (!(interactiveRunner instanceof InteractiveBenchmarkRunner)) {
-        throw Error(
-            "window.BenchmarkRunner must be "+
-            "a subclass of InteractiveBenchmarkRunner");
+        throw Error("window.BenchmarkRunner must be " + "a subclass of InteractiveBenchmarkRunner");
```
Comment (rniwa) [inline resources/interactive.mjs:222]: Looks like this can just be a single string literal?

**Review by rniwa — APPROVED** [review]

File: resources/interactive.mjs (hunk 8 above)
Comment (flashdesignory) [reply resources/interactive.mjs:222]: nice catch - I can address that with the template literals update in a follow up pr

**Pull request merged**